kernels/volk/volk_16ic_deinterleave_16i_x2.h

```diff
@@ -375,4 +375,45 @@ static inline void volk_16ic_deinterleave_16i_x2_u_avx2(int16_t* iBuffer,
 }
 #endif /* LV_HAVE_AVX2 */
 
+#ifdef LV_HAVE_RVV
+#include <riscv_vector.h>
+
+static inline void volk_16ic_deinterleave_16i_x2_rvv(int16_t* iBuffer,
+                                                     int16_t* qBuffer,
+                                                     const lv_16sc_t* complexVector,
+                                                     unsigned int num_points)
+{
+    size_t n = num_points;
+    for (size_t vl; n > 0; n -= vl, complexVector += vl, iBuffer += vl, qBuffer += vl) {
+        vl = __riscv_vsetvl_e16m4(n);
+        vuint32m8_t vc = __riscv_vle32_v_u32m8((const uint32_t*)complexVector, vl);
+        vuint16m4_t vr = __riscv_vnsrl(vc, 0, vl);
+        vuint16m4_t vi = __riscv_vnsrl(vc, 16, vl);
+        __riscv_vse16((uint16_t*)iBuffer, vr, vl);
+        __riscv_vse16((uint16_t*)qBuffer, vi, vl);
+    }
+}
+#endif /*LV_HAVE_RVV*/
+
+#ifdef LV_HAVE_RVVSEG
+#include <riscv_vector.h>
+
+static inline void volk_16ic_deinterleave_16i_x2_rvvseg(int16_t* iBuffer,
+                                                        int16_t* qBuffer,
+                                                        const lv_16sc_t* complexVector,
+                                                        unsigned int num_points)
+{
+    size_t n = num_points;
+    for (size_t vl; n > 0; n -= vl, complexVector += vl, iBuffer += vl, qBuffer += vl) {
```

Comment:
Can you default initialize `vl`? `size_t vl = 0`?

Reply:
I could, but this is how the reference implementations currently write it:
https://github.com/riscv-non-isa/rvv-intrinsic-doc/blob/main/examples/rvv_branch.c#L15

Reply:
OK. We can leave it as is. It's funny though that you contributed to this example and point to it now.

Reply:
Oh, true, but I only cleaned the examples up a bit, this paradigm was there before.